woocommerce/woocommerce-ios
# [Woo POS] Initial UI scaffolding

Closes: https://github.com/woocommerce/woocommerce-ios/issues/12701

https://github.com/woocommerce/woocommerce-ios/pull/12685 needs to be merged first.


## Description
This PR builds up on top of the initial entry point and updates the following:

- Updates the `PointOfSaleDashboardView` to be presented as a non-modal view.
- Updates `BetaFeature` so the experimental toggle and feature row are only shown in iPads, previously we would just present a modal warning it was not supported.
- Implements some general structure from the wireframes ( p91TBi-bcW-p2 ) by adding a `ProductGridView`, an `OrderView`, and a `ProductRowView`. All of them entirely un-styled yet. Those can be tackled separately.
- Adds a dummy `ProductFactory` util and a `Product` model so we can render something from a data layer.
- Adds top navigation with 3 initial toolbar buttons, only `Exit POS` is functional at the moment, dismissing the POS view and moving back to the Woo App.
- Adds a callback when we allocate and deallocate the POS dashboard from memory by toggling the Woo app bottom tab bar visibility when appropriate.

![Simulator Screen Recording - iPad Pro (12 9-inch) (6th generation) - 2024-05-12 at 18 50 42](https://github.com/woocommerce/woocommerce-ios/assets/3812076/036b577c-3f7e-4adc-989c-b2fd548bf3a4)

## Testing instructions
* On iPhone, navigate to Menu > Settings > Experimental features and observe the POS toggle is not there. The settings row does not appear either.
* On Tablet, with the experimental toggle enabled (switch stores if the row doesn't appear), open the POS and observe that:
  * The bottom navigation bar disappears, and the top navigation bar consists of 3 buttons
  * By tapping on "Exit POS" we're brought back to the Woo app and the bottom navigation re-appears and works as expected. Navigate through the items, and back and forth between the app and the pos to confirm works as expected.
